### PR TITLE
solution

### PR DESCRIPTION
**Pull request by abcdesign1**
[DEMO LINK](https://vadym4che.github.io/react_keyboard/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,35 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+export class App extends React.Component {
+  state = {
```
Comment (sergii-nosachenko) [inline src/App.tsx:4]: Set state to be Readonly

File: src/App.tsx [hunk 2]
```diff
@@ -1,7 +1,35 @@
...
+  componentDidMount() {
+    document.addEventListener('keyup', this.handleKeyUp);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.handleKeyUp);
+  }
+
+  handleKeyUp = (event: KeyboardEvent) => {
+    const k = event.key;
```
Comment (sergii-nosachenko) [inline src/App.tsx:17]: What is `k`? Use destructuring.

File: src/App.tsx [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
+    this.setState({ pressedKey: k });
+  };
+
+  render() {
+    const { pressedKey: key } = this.state;
+
+    return (
+      <div className="App">
+        <p className="App__message">
+          {key !== null
```
Comment (sergii-nosachenko) [inline src/App.tsx:28]:
```suggestion
          {pressedKey
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,7 +1,35 @@
...
+  }
+
+  handleKeyUp = (event: KeyboardEvent) => {
+    const k = event.key;
+
+    this.setState({ pressedKey: k });
+  };
+
+  render() {
+    const { pressedKey: key } = this.state;
```
Comment (sergii-nosachenko) [inline src/App.tsx:23]: You don't need to rename variable

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (vadym4che) [reply src/App.tsx:4]: fixed

File: src/App.tsx (hunk 2 above)
Comment (vadym4che) [reply src/App.tsx:17]: fixed

File: src/App.tsx (hunk 3 above)
Comment (vadym4che) [reply src/App.tsx:28]: fixed

File: src/App.tsx (hunk 4 above)
Comment (vadym4che) [reply src/App.tsx:23]: fixed

**Pull request closed without merging**